### PR TITLE
cli: add recommendation commands.

### PR DESCRIPTION
**Pull request by jrasell**
Adds new CLI commands for applying, dismissing and detailing Nomad
recommendations under a new top level recommendation command.

### CODE REVIEW

File: command/recommendation_apply.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+  ` + generalOptionsUsage() + `
+
+Recommendation Apply Options:
+
+  -detach
+    Return immediately instead of entering monitor mode. After applying a
+    recommendation, the evaluation ID will be printed to the screen, which can
+    be used to examine the evaluation using the eval-status command. If applying
+    multiple recommendations, this value will always be true.
```
Comment (cgbaker) [inline command/recommendation_apply.go:37]:
```suggestion
    recommendations for multiple jobs, this value will always be true.
```

File: command/recommendation_apply.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		length = fullId
+	}
+	mon := newMonitor(r.Ui, client, length)
+	return mon.monitor(resp.UpdatedJobs[0].EvalID)
+}
+
+func (r *RecommendationApplyCommand) outputApplyErrors(errs []*api.SingleRecommendationApplyError) {
+	output := []string{"IDs|Job ID|Error"}
+	for _, err := range errs {
+		output = append(output, fmt.Sprintf("%s|%s|%s", err.Recommendations, err.JobID, err.Error))
```
Comment (cgbaker) [inline command/recommendation_apply.go:144]: these errors come from Job.Register; I don't know how they're formatted, but I suspect it's multiline and gross.
not sure what else we can do here.

File: command/recommendation_list.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+			return 1
+		}
+		r.Ui.Output(out)
+		return 0
+	}
+
+	// Create the output table header.
+	output := []string{"ID|"}
+
+	// If the operator is using the namespace wildcard option, add this header.
```
Comment (cgbaker) [inline command/recommendation_list.go:151]: 👍

**Review by cgbaker — APPROVED** [review]
👍

**Review by lgfa29 — APPROVED** [review]
:shipit:

File: command/recommendation_apply.go (hunk 2 above)
Comment (jrasell) [reply command/recommendation_apply.go:144]: Is it worth running the error string through a regex replace, removing newlines and replacing with commas?

Comment (cgbaker) [reply command/recommendation_apply.go:144]: ugh, maybe? i think that's just gonna make them gross in a different direction.
maybe just let them hang out in all of their glory. it really only gets bad if there's multiple job updates with errors.

**Pull request merged**